src/screens/TopicPageScreen/elements/BottomSheetFollow.tsx

```diff
@@ -24,25 +22,25 @@ interface BottomSheetFollowProps {
   memberCount: number;
   setMemberCount: (memberCount: number) => void;
   onClose: () => void;
-  getTopicDetail: (topicName: string) => void;
   followType: Follow;
   setFollowType: (followType: Follow) => void;
+  topicId: string;
 }
 
 const BottomSheetFollow = forwardRef((props: BottomSheetFollowProps, ref: Ref<RBSheet>) => {
   const {
+    topicId,
     topicName,
     memberCount,
     setMemberCount,
     isFollow,
     setIsFollow,
     followType,
     setFollowType,
-    getTopicDetail,
     onClose
   } = props;
-  const [, discoveryDispatch] = (React.useContext(Context) as any)?.discovery;
   const {followTopic} = useChatClientHook();
+  const {updateFollowTopicDiscoveryContext} = useDiscovery();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a `topicId` prop to the `BottomSheetFollowProps` interface and uses it in the component. However, there's no validation or error handling for this prop. If `topicId` is not provided or is invalid, it could lead to unexpected behavior or errors. Consider adding prop validation or error handling.

```diff
+  if (!topicId) {
+    throw new Error('Topic ID is required');
+  }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->